src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1286,36 +1289,60 @@ public void Paste()
                 AddToSelection(newNote);
             }
 
-            NodeModel start;
-            NodeModel end;
+            ModelBase start;
+            ModelBase end;
             var newConnectors =
                 from c in connectors
 
                 // If the guid is in nodeLookup, then we connect to the new pasted node. Otherwise we
                 // re-connect to the original.
                 let startNode =
-                    nodeLookup.TryGetValue(c.Start.Owner.GUID, out start)
+                    modelLookup.TryGetValue(c.Start.Owner.GUID, out start)
                         ? start
                         : CurrentWorkspace.Nodes.FirstOrDefault(x => x.GUID == c.Start.Owner.GUID)
                 let endNode =
-                    nodeLookup.TryGetValue(c.End.Owner.GUID, out end)
+                    modelLookup.TryGetValue(c.End.Owner.GUID, out end)
                         ? end
                         : CurrentWorkspace.Nodes.FirstOrDefault(x => x.GUID == c.End.Owner.GUID)
 
                 // Don't make a connector if either end is null.
                 where startNode != null && endNode != null
                 select
-                    ConnectorModel.Make(startNode, endNode, c.Start.Index, c.End.Index);
+                    ConnectorModel.Make(startNode as NodeModel, endNode as NodeModel, c.Start.Index, c.End.Index);
```

Comment:
The result of your `as` could potentially be null.  It makes more sense to catch this case by doing the `as` in the `let` statements above.  Regardless of your approach, try and catch this case earlier rather than passing potentially nasty data down to `ConnectorModel.Make`.